src/Concerns/ConfiguresProviders.php

```diff
@@ -41,4 +41,19 @@ public function usingProviderConfig(array $config): self
 
         return $this;
     }
+
+    public function getModel(): string
+    {
+        return $this->model;
+    }
+
+    public function getProviderKey(): string
+    {
+        return $this->providerKey;
+    }
+
+    public function getProviderEnum(): ProviderEnum
+    {
+        return ProviderEnum::from($this->providerKey);
+    }
```

Comment:
I'm thinking that we only return the key, since both can be used. There will ALWAYS be a provider key, but not always a provider Enum because users can register custom providers that aren't going to be in the Enum.

```suggestion
    public function providerKey(): string
    {
        return $this->providerKey;
    }
```